gradoop-flink/src/main/java/org/gradoop/flink/io/impl/csv/indexed/functions/IndexedCSVFileFormat.java

```diff
@@ -136,103 +113,39 @@ public IndexedCSVFileFormat(Path outputPath, String recordDelimiter, String fiel
-    this.labelsToWriter = new HashMap<>();
   }
 
   /**
-   * Check the label of a tuple to identify the name of the according output file.
-   * If there is not already a writer for the label, it will map it to the tuple.
+   * Sets the charset with which the CSV strings are written to the file.
+   * If not specified, the output format uses the systems default character encoding.
    *
-   * @param fileName Name of the file for the tuple.
-   * @param tuple Tuple that will be mapped to writer.
-   * @throws IOException - Throne if creating writer or output stream fails.
+   * @param charsetName The name of charset to use for encoding the output.
    */
-  public void mapWriter(Tuple tuple, String fileName) throws IOException {
-    if (labelsToWriter.containsKey(fileName)) {
-      writeToCSV(tuple, labelsToWriter.get(fileName));
-    } else {
-      FSDataOutputStream stream = super.getAndCreateFileStream(fileName);
-      Writer wrt = this.charsetName == null ? new OutputStreamWriter(
-          new BufferedOutputStream(stream, 4096), "UTF8") :
-            new OutputStreamWriter(new BufferedOutputStream(stream, 4096), this.charsetName);
-      labelsToWriter.put(fileName, wrt);
-      writeToCSV(tuple, wrt);
-    }
+  public void setCharsetName(String charsetName) {
+    this.charsetName = charsetName;
   }
 
-  /**
-   * Write every field of the tuple in a csv file.
-   * Separates record via record delimiter and every field in a record
-   * with field delimiters.
-   *
-   * @param tuple the tuple to write in the output file
-   * @param writer the writer for the tuple
-   * @throws IOException - Thrown, if the records could not be added to to an I/O problem.
-   */
-  public void writeToCSV(Tuple tuple, Writer writer) throws IOException {
-
-    int numFields = tuple.getArity();
-    for (int i = 0; i < numFields; i++) {
-      Object v = tuple.getField(i);
-      if (v != null) {
-        if (i != 0) {
-          writer.write(this.fieldDelimiter);
-        }
-
-        if (quoteStrings) {
-          if (v instanceof String || v instanceof StringValue) {
-            writer.write('"');
-            writer.write(v.toString());
-            writer.write('"');
-          } else {
-            writer.write(v.toString());
-          }
-        } else {
-          writer.write(v.toString());
-        }
-      } else {
-        if (this.allowNullValues) {
-          if (i != 0) {
-            writer.write(this.fieldDelimiter);
-          }
-        } else {
-          throw new RuntimeException("Cannot write tuple with <null> value at position: " + i);
-        }
-      }
+  @Override
+  protected OutputFormat<T> createFormatForDirectory(Path directory) throws IOException {
+    CsvOutputFormat<T> format = new CsvOutputFormat<>(directory, recordDelimiter, fieldDelimiter);
+    // no overwrite allows multiple output formats to write concurrently
+    format.setWriteMode(FileSystem.WriteMode.NO_OVERWRITE);
```

Comment:
Setting the overwrite here to a fixed value (and not using the `writeMode` attribute of the parent class) , the `overwrite` flag of the Sink is not working. Please add an IOException or UnsupportedOperationException to the Sink if an user sets the overwrite flag to true. Or make sure that this functionality works properly.

Reply:
This is for the output formats of each worker/label. `OVERWRITE` would delete the whole label directory and files of different workers when a worker encounters the first element of a label. This should not influence the global overwrite behaviour. I will improve the comment.

Reply:
OK. How I said - either fix the overwriting or add the Exception.